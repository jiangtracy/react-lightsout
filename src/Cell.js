import React from "react";
import "./Cell.css";

/** A single cell on the board.
 *
 * This has no state --- just 3 props:
 *
 * - flipCellsAroundMe: a function rec'd from the board which flips this
 *      cell and the cells around of it
 *
 * - isLit: boolean, is this cell lit?
 * 
 * - id: id that the cell should render to
 *
 * This handles clicks --- by calling flipCellsAroundMe
 *
 **/

function Cell({ flipCellsAroundMe, isLit, id }) {
  const classes = `Cell ${isLit ? "Cell-lit" : ""}`;
  return <td className={classes} id={id} onClick={flipCellsAroundMe} />;
}

export default Cell;
