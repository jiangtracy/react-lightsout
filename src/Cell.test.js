import React from "react";
import { render } from "@testing-library/react";
import Cell from "./Cell";

it("matches snapshot", function() {
  const {container} = render(<Cell isLit={true}/>);
  expect(container).toMatchSnapshot();
});

it("generate the correct color", function() {
  const {container, debug} = render(<Cell isLit={true}/>);
  debug(container);
  expect(container.querySelector("td")).toHaveClass("Cell-lit");
});





