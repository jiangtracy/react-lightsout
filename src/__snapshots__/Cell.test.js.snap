// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <td
    class="Cell Cell-lit"
  />
</div>
`;
